### DISCUSSION TITLE
Red squiggy line under include statement using a VCPKG dependency....

### DISCUSSION THREAD

**Discussion by awa5114** [microsoft/vscode-cpptools#12855] [Q&A]
I've found a couple of threads that kind of/sort of discuss the issue I'm having, but none of them helped:
- https://github.com/microsoft/vscode-cpptools/issues/4002
- https://github.com/microsoft/vscode-cpptools/discussions/10261

I have a pretty simple project resulting from following the tutorial in the Microsoft [Docs](https://learn.microsoft.com/en-gb/vcpkg/get_started/get-started-vscode?pivots=shell-powershell) to get a project using vcpkg to work with VS Code. It consists of the following files:

**CMakeLists.txt**

```
cmake_minimum_required(VERSION 3.10)
project(GdalCppProject)

message("VARIABLES")
message("\tCMAKE_MAKE_PROGRAM: ${CMAKE_MAKE_PROGRAM}")
message("\tCMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}")

find_package(fmt CONFIG REQUIRED)
add_executable(GdalCppProject GdalCppProject.cpp)
target_link_libraries(GdalCppProject PRIVATE fmt::fmt)
```

**CMakePresets.json**

```
{
    "version": 2,
    "configurePresets": [
        {
            "name": "vcpkg",
            "generator": "Ninja",
            "binaryDir": "${sourceDir}/build",
            "cacheVariables": {
                "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
            },
            "environment": {
                "VCPKG_ROOT": "C:/Program Files/Microsoft Visual Studio/2022/Community/VC/vcpkg"
            }
        }
    ]
}
```

GdalCppProject.cpp

```
#include <fmt/core.h>

int main()
{
    std::string x = "Msddsdssdars";
    fmt::print("Hello {}!\n", x);
    return 0;
}
```

vcpkg-configuration.json

```
{
  "default-registry": {
    "kind": "git",
    "baseline": "6f1ddd6b6878e7e66fcc35c65ba1d8feec2e01f8",
    "repository": "https://github.com/microsoft/vcpkg"
  },
  "registries": [
    {
      "kind": "artifact",
      "location": "https://github.com/microsoft/vcpkg-ce-catalog/archive/refs/heads/main.zip",
      "name": "microsoft"
    }
  ]
}
```

vcpkg.json

```
{
  "dependencies": [
    "fmt"
  ]
}
```

The project builds, runs and debugs fine in VS Code, but line 1 of **GdalCppProject.cpp** is underlined with a red squiggly line and I get the following problem:

> Cannot open include file: 'fmt/core.h': No such file or directory

How do I get the extension to recognize dependencies managed with `vcpkg`? The only C++ relevant extensions I have activated are the C/C++ and the CMake Tools extensions by Microsoft.

**Comment by Kyr27** [comment]
Not really the answer, but a workaround you can try, and the recommended way of doing it to use fmt is to download it using CMake FetchContent, here's how i have it setup for one of my projects, CMakeLists.txt:

# Fetch fmt library and make it available
include(FetchContent)
FetchContent_Declare(
    fmt
    GIT_REPOSITORY https://github.com/fmtlib/fmt
    GIT_TAG 11.0.2
)
FetchContent_MakeAvailable(fmt)

add_executable(${PROJECT_NAME} 
src/main.cpp

# Make sure the following are below your add_executable
# Link the fmt library and include its headers
target_link_libraries(${PROJECT_NAME} fmt::fmt)

# Include fmt headers directory
target_include_directories(${PROJECT_NAME} PRIVATE ${fmt_SOURCE_DIR}/include)
